libs/coin-evm/src/prepareTransaction.ts

```diff
@@ -127,7 +127,24 @@ export const prepareTransaction = async (
 ): Promise<EvmTransaction> => {
   const { currency } = account;
   // Get the current network status fees
-  const feeData = await getFeesEstimation(currency);
+  const feeData = await (async () => {
+    if (!transaction.feesStrategy) {
+      return await getFeesEstimation(currency);
```

Comment:
Except in try/catch, return await is not necessary I think.
Also is this early return really necessary ? I think not having it is having the same result at line 143 because you'll get an undefined `gasOption` and return the `getFeesEstimation` instead ?

Reply:
> Also is this early return really necessary ? I think not having it is having the same result at line 143 because you'll get an undefined gasOption and return the getFeesEstimation instead ?

That would be true in JS (or loosely typed TS) code, but here the `GasOptions` type only has `slow`, `medium` and `fast` valid keys. `null` or `undefined` are not valid keys (and I don't think they should be).

This if statement basically acts as a type guard / type narrowing logic, ensuring that when we try to access a `gasOption` field, we are using a valid key (i.e: not a `null` or `undefined` one).

For example, here is what TS says when we remove this if statement:
![TS error](https://github.com/LedgerHQ/ledger-live/assets/9203826/79dc3348-f00c-4d99-89e6-e1321f7311b1)

An alternative could be to remove this if statement and update line 143 to something like:
```ts
    const gasOption = transaction.feesStrategy
      ? transaction.gasOptions?.[transaction.feesStrategy]
      : undefined;
```

I'm not a big fan of this here though, I feel like it makes the code less readable / straightforward

Reply:
I think you can fix this by casting this way:

```typescript
const gasOption = transaction.gasOptions?.[transaction.feeStrategy as keyof typeof transaction.gasOptions];
```
Not super good looking by hey, typescript right ? 🤷‍♂️